pandas/core/groupby/groupby.py

```diff
@@ -692,7 +692,7 @@ def __iter__(self):
         Generator yielding sequence of (name, subsetted object)
         for each group
         """
-        return self.grouper.get_iterator(self.obj, axis=self.axis)
+        return self.grouper.get_iterator(self._selected_obj, axis=self.axis)
```

Comment:
This was rather surprising that I had to change here, but getting rid of the block code seemed to change the behavior of the following snippet:

```python
In [14]: df = pd.DataFrame([[1, 2, 3], [4, 5, 6]], columns=["g", "a", "a"])

In [15]: df.groupby("g").transform("first")
Out[15]:
   a  a
0  2  3
1  5  6
```

Without referencing `self._selected_obj` this would still include the grouping as part of the output (this change might close some open issues; will check later)

Reply:
That is surprising.  Is it possible that it is wrong now but untested?  (i.e. could be fixed independently)